deploy-service/common/src/main/java/com/pinterest/deployservice/handler/GoalAnalyst.java

```diff
@@ -277,6 +277,25 @@ AgentState proposeNewAgentState(PingReportBean report, AgentBean agent) {
     // This is intended to be used for update agent record, not for goal,
     // We populate all the fields, since this could be used for insertOrUpdate as well
     AgentBean genUpdateBeanByReport(PingReportBean report, AgentBean agent) {
+        AgentState newAgentState = proposeNewAgentState(report, agent);
+
+        // Don't update if not changed.
+        if (agent != null && agent.getHost_id() == host_id && agent.getDeploy_id() == report.getDeployId() &&
```

Comment:
Is this checks are fine? I mean, they are Strings, why not use equals instead?

Reply:
yup, this won't work. Working on changing it.